src/main/kotlin/skytils/skytilsmod/features/impl/mining/MiningFeatures.kt

```diff
@@ -185,52 +187,37 @@ class MiningFeatures {
                 waypointChatMessage(xzMatcher.group("x"), "100", xzMatcher.group("z"))
                 return
             }
-            if (unformatted.contains("\$SBECHWP:") || unformatted.contains("\$DSMCHWP:")) {
-                val sub = unformatted.substring(unformatted.lastIndexOf(":") + 1)
+
+            /**
+             * Checks for the format used in DSM and SBE
+             * $DSMCHWP:Mines of Divan@-673,117,426 ✔
+             * $SBECHWP:Khazad-dûm@-292,63,281 ✔
+             * $asdf:Khazad-dûm@-292,63,281 ❌
+             * $SBECHWP:Khazad-dûm@asdf,asdf,asdf ❌
+             */
+            if (SBE_DSM_PATTERN.matcher(unformatted).matches()) {
+                val sub = unformatted.split(Regex("\\\$(SBECHWP\\b|DSMCHWP):"))[1]
                 val parts = sub.split("@")
+                val location = parts[0]
                 val coords = parts[1].split(",")
```

Comment:
the point of using a regex was so you can use capture groups to get the coords rather than calling a bunch of splits